include/seqan3/alignment/all.hpp

```diff
@@ -10,20 +10,295 @@
+ * Types that model seqan3::ScoringScheme can (but don't need to!) inherit from it.</i></small>
+ *
+ * ### Scoring gaps
+ *
+ * Gaps are usually not scored on a per-character-basis, because it is widely recognised that the likelihood of
+ * `n` consecutive gaps is much higher than that of `n` individual gaps. This is based on the assumption that
+ * one biological event often introduces more than one gap at a time and that single character gaps are not common
+ * due to other biological factors like frame preservation in protein-coding sequences.
+ *
+ * Therefore, the gaps are not handled by the scoring scheme. SeqAn offers the additional seqan3::gap_scheme
+ * which can be used to set the scores for opening or extending a gap.
+ *
+ * The gap scheme can be configured with the seqan3::align_cfg::gap element. If the configuration is not specified,
+ * the algorithm uses edit distance scores (`-1`) for deletion/insertion.
+ *
+ * ## Computing banded alignments
+ *
+ * SeqAn offers the computation of banded alignments to reduce the running time of the algorithm. This can be
+ * helpful if the region in which the optimal alignment exists is known a priori. To specify the banded alignment
+ * the developer can use the seqan3::align_cfg::band option initialised with a seqan3::static_band.
+ * The seqan3::static_band is constructed with a lower bound and an upper bound. The upper bound must always be greater
+ * than or equal to the lower bound. To position the band, imagine a rectangle where the first sequence is written on
+ * top and the second sequence along the left vertical side. A negative bound implies a start of the band within the
+ * vertical part while a positive bound implies a start within the top part of this rectangle at the respective
+ * position.
+ *
+ * ## Global and local alignments
+ *
+ * The standard global and local alignments can be configured using the configuration seqan3::align_cfg::mode.
+ * The global alignment can be further refined using the seqan3::align_cfg::sequence_ends configuration.
+ * This configuration allows to enable, respectively disable the scoring of leading and trailing gaps at the respective
+ * sequence ends. This option is not available for the local alignment where scoring gaps at the ends of the sequences
+ * is always disabled.
+ *
+ * ## Computing edit distance
+ *
+ * A special implementation of the global sequence alignment is the edit distance. This algorithm computes merely
+ * the number of edit operations (substitutions, insertions and deletions). It can only be computed with a scoring
+ * scheme initialised with `0` for matching letters and `-1` for mismatching letters and a gap scheme with `-1` for
+ * inserting or deleting a gap. Currently, it is only available for nucleotide scoring schemes. There is a special
+ * edit distance shortcut config seqan3::align_cfg::edit which is initialised with the correct scoring and gap scheme.
```

Comment:
Why do we have an extra config element? Shouldn't it just be chosen when the other config options allow it?

Reply:
It is a predefined shortcut. so instead of writing `align_cfg::mode{global} | align_cfg::scoring{...} | align_cfg::gap{...}` one can use `align_cfg::edit` which is a constexpr inline variable.